tools/taskcluster/crashreporter/symbolstore.py

```diff
@@ -0,0 +1,714 @@
+            return [self.dump_syms] + arch.split() + ['-g', dsymbundle, file]
+        return Dumper.dump_syms_cmdline(self, file, arch)
+
+    def GenerateDSYM(self, file):
+        """dump_syms on Mac needs to be run on a dSYM bundle produced
+        by dsymutil(1), so run dsymutil here and pass the bundle name
+        down to the superclass method instead."""
+        t_start = time.time()
+        print("Running Mac pre-processing on file: %s" % (file,),
+              file=sys.stdout)
+
+        dsymbundle = file + ".dSYM"
+        if os.path.exists(dsymbundle):
+            shutil.rmtree(dsymbundle)
+        dsymutil = buildconfig.substs['DSYMUTIL']
+        # dsymutil takes --arch=foo instead of -a foo like everything else
+        try:
+            cmd = ([dsymutil] +
+                   [a.replace('-a ', '--arch=') for a in self.archs if a] +
+                   [file])
+            print(' '.join(cmd), file=sys.stdout)
+            subprocess.check_call(cmd, stdout=open(os.devnull, 'w'))
+        except subprocess.CalledProcessError as e:
+            print('Error running dsymutil: %s' % str(e), file=sys.stderr)
+            raise
+
+        if not os.path.exists(dsymbundle):
+            # dsymutil won't produce a .dSYM for files without symbols
+            print("No symbols found in file: %s" % (file,), file=sys.stderr)
+            return False
+
+        elapsed = time.time() - t_start
+        print('Finished processing %s in %.2fs' % (file, elapsed),
+              file=sys.stdout)
+        return dsymbundle
+
+    def CopyDebug(self, file, debug_file, guid, code_file, code_id):
+        """ProcessFile has already produced a dSYM bundle, so we should just
+        copy that to the destination directory. However, we'll package it
+        into a .tar.bz2 because the debug symbols are pretty huge, and
+        also because it's a bundle, so it's a directory. |file| here is the
+        the original filename."""
+        dsymbundle = file + '.dSYM'
+        rel_path = os.path.join(debug_file,
+                                guid,
+                                os.path.basename(dsymbundle) + ".tar.bz2")
+        full_path = os.path.abspath(os.path.join(self.symbol_path,
+                                                  rel_path))
+        success = subprocess.call(["tar", "cjf", full_path, os.path.basename(dsymbundle)],
+                                  cwd=os.path.dirname(dsymbundle),
+                                  stdout=open(os.devnull, 'w'), stderr=subprocess.STDOUT)
+        if success == 0 and os.path.exists(full_path):
+            print(rel_path)
+
+def Upload_Symbol(zip_file):
+    print("Uploading symbol file '{0}' to '{1}'".format(zip_file, DEFAULT_SYMBOL_URL), file=sys.stdout)
+
+    # Fetch the symbol server token from Taskcluster secrets
+    secrets_url = "http://taskcluster/secrets/v1/secret/{}".format("project/firefoxreality/symbols-token")
+    res = requests.get(secrets_url)
```

Comment:
Please help confirm if it is correct to fetch the token from the Secret. I can't get the token in my local, I suppose it probably could be gotten in our Taskcluster.